cms/static/js/views/pages/container.js

```diff
@@ -535,48 +523,53 @@ function($, _, Backbone, gettext, BasePage,
             });
         },
 
-        duplicateComponent: function(xblockElement) {
-            // A placeholder element is created in the correct location for the duplicate xblock
-            // and then onNewXBlock will replace it with a rendering of the xblock. Note that
-            // for xblocks that can't be replaced inline, the entire parent will be refreshed.
-            var self = this,
-                parentElement = self.findXBlockElement(xblockElement.parent()),
-                scrollOffset = ViewUtils.getScrollOffset(xblockElement),
-                $placeholderEl = $(self.createPlaceholderElement()),
-                placeholderElement;
-
-            placeholderElement = $placeholderEl.insertAfter(xblockElement);
-            XBlockUtils.duplicateXBlock(xblockElement, parentElement)
-                .done(function(data) {
-                    self.onNewXBlock(placeholderElement, scrollOffset, true, data);
-                })
-                .fail(function() {
-                    // Remove the placeholder if the update failed
-                    placeholderElement.remove();
-                });
-        },
-
         duplicateXBlock: function(event) {
             event.preventDefault();
+            if (this.options.isIframeEmbed) {
+                window.parent.postMessage(
+                    {
+                        type: 'duplicateXBlock',
+                        payload: {}
+                    }, document.referrer
+                );
+                return
+            }
```

Comment:
[optional]: At this stage we do not expect fully working functionality, but it seems to me that it would be great to provide minimal working functionality. Since now after any action we get the error like `Failed to execute 'postMessage' on 'Window': Invalid target origin '' in a call to 'postMessage'`. I suggest catching such errors and not blocking the execution of the JS code flow. My proposal applies to all syntactic constructions in this file that are related to `postMessage`. What are your thoughts?

```suggestion
            try {
                if (this.options.isIframeEmbed) {
                    window.parent.postMessage(
                        {
                            type: 'duplicateXBlock',
                            payload: {}
                        }, document.referrer
                    );
                    return
                }
            } catch (e) {
                console.error(e);
            }
```

Reply:
updated